File: easy/1758-minimum-changes-to-make-alternating-binary-string/solution.py
class Solution:
    def minOperations(self, s: str) -> int:
        """
        1758. Minimum Changes To Make Alternating Binary String
        Time: O(n)
        Space: O(1)
        """
        change_start0 = 0
        change_start1 = 0

        for i, ch in enumerate(s):
            if i % 2 == 0:
                if ch != '0':
                    change_start0 += 1
                if ch != '1':
                    change_start1 += 1
            else:
                if ch != '1':
                    change_start0 += 1
                if ch != '0':
                    change_start1 += 1

        return min(change_start0, change_start1)

File: easy/1758-minimum-changes-to-make-alternating-binary-string/solution.java
/**
 * 1758. Minimum Changes To Make Alternating Binary String
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int minOperations(String s) {
        int changeStart0 = 0;
        int changeStart1 = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);

            if (i % 2 == 0) {
                if (ch != '0') changeStart0++;
                if (ch != '1') changeStart1++;
            } else {
                if (ch != '1') changeStart0++;
                if (ch != '0') changeStart1++;
            }
        }

        return Math.min(changeStart0, changeStart1);
    }
}
